src/main.ts

```diff
@@ -211,17 +210,31 @@ function populateEnemyId() {
 }
 
 function challengeUser() {
-  //TODO: Add error message if no user Id specified and success message when connected successfully
-  var playerId = getEnemyId();
-  if (senderConnection == null || senderConnection == undefined) {
-    senderConnection = myself.connect(playerId, { reliable: true });
-    if (connection == null || connection == undefined) {
-      myScore = leftScore;
-      enemyScore = rightScore;
-      myPaddle = leftPaddle;
-      enemyPaddle = rightPaddle;
-      iAmHost = true;
-      startPlaying = true;
+  if (connection == null || connection == undefined) {
+    var enemyId = getEnemyId();
+    if(enemyId == null || enemyId == undefined) {
+      alert('enemy id is empty');
+    } else {
+      var retries = 0;
+      var isSuccess = false;
+      while(retries < 3 && !isSuccess) {
+        connection = myself.connect(enemyId, { reliable: true });
+        if (connection != null) {
+          alert('successfully connected to peer ' + connection.peer);
```

Comment:
Alert on successful connection may interfere with gameplay as it needs to be closed in order to get control of the keyboard again, could this be removed?

Reply:
merged latest develop and removed the success message, I will add a note to the issue